### DISCUSSION TITLE
People Search

### DISCUSSION THREAD

**Discussion by ITSMillsaps** [microsoft-search/pnp-modern-search#4158] [Q&A]
We are trying to create an employee directory using PNP search but we want to limit it to only certain groups and not everyone.  Right now it returns everyone including service accounts, students, etc.  Need to limit to faculty and staff only.  I know this is done in query template but cannot find the right query to make this work.

**Comment by kasperbolarsen** [comment]
You might find this blog post of mine useful:https://www.m365thinking.com/post/sharepoint-people-search-how-to-clean-up-your-results-by-excluding-accounts

In general you will have to look into which of the user profile properties that will enable you to segregate your various groups. 
If no combination of the existing properties really solve the issue,  you might have to add another custom User Profile Application property,  but that should be the last resort

**Comment by ITSMillsaps** [comment]
Thanks for reply

What I am looking for exclude any account that user does not have a departent or I can use if the user does not have a title.  This would exclude a majority of what I need excluded.



David Mattox
Office: 601.974.1149
***@***.******@***.***/> | www.millsaps.edu<http://millsaps.edu/>



________________________________

**Comment by ITSMillsaps** [comment]
How would I exclude user with no department since I see department is part of the profile?

David Mattox
Office: 601.974.1149
***@***.******@***.***/> | www.millsaps.edu<http://millsaps.edu/>

**Reply by jblueml** [reply]
You can use `Department:*` to retrieve only profiles that contain a value in this managed property.

**Reply by kasperbolarsen** [reply]
Or owstaxIdSPShDepartment:"#8ed8c9ea-7052-4c1d-a4d7-b9c10bffea6f"   ( only show the account having a value in Department that matches one of the values in the Term set Department)
"8ed8c9ea-7052-4c1d-a4d7-b9c10bffea6f" is the guid for the root node in the Department Term set
